### PR TITLE
cicd: new release and change log process

### PR DESCRIPTION
**Pull request by ldelossa**
this commit introduces automated change log generation and github
workflows to support it

Signed-off-by: ldelossa <ldelossa@redhat.com>

### CODE REVIEW

File: .chglog/config.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+style: github
+template: CHANGELOG.tpl.md
+info:
+  title: CHANGELOG
+  repository_url: https://github.com/ldelossa/gcl
```
Comment (hdonnay) [inline .chglog/config.yml:5]: Seems wrong

File: .chglog/config.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  commit_groups:
+    group_by: Scope
+  header:
+    pattern: '^(.*):\s*(.*)$'
+    pattern_maps:
+      - Scope
+      - Subject
+  issues:
+    prefix: 
+      - "#"
```
Comment (hdonnay) [inline .chglog/config.yml:18]: Is this right if we're not using github issues?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -4,6 +4,18 @@ name: CI
...
 jobs:
+  commit-check:
+    name: Commit Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: commit check
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: |
+            ^(.*):\s*(.*)$
```
Comment (hdonnay) [inline .github/workflows/main.yml:15]: Does this run on just header or the entire message?

File: .github/workflows/prepare-release.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+---
+name: Prepare Release
+# Prepare Release looks for a tag with a "prepare" prefix.
+# If encountered an automated changelog pull request will be opened against the repository.
+# If the tag does not contain alpha, beta, or rc test artifacts will be created for QE testing.
```
Comment (hdonnay) [inline .github/workflows/prepare-release.yml:5]: This comment seems incorrect. I can't see how this is triggered from a tag

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -1,11 +1,63 @@
...
+          chglog="${chglog//$'\n'/'%0A'}"
+          chglog="${chglog//$'\r'/'%0D'}"
+          cat <<.
+          ::set-env name=CHANGELOG::${chglog}
+          .
+      - name: ChangeLogSanityCheck
+        run: |
+          cat <<.
+          ${{ env.CHANGELOG }}
+          .
```
Comment (hdonnay) [inline .github/workflows/release.yml:45]: This seems superfluous

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -1,11 +1,63 @@
...
+          tag_name: ${{ github.ref }}
+          release_name: ${{ env.VERSION}} Release
+          body: |
+            ${{ env.CHANGELOG }}
+          prerelease: ${{ contains(env.VERSION, 'alpha') || contains(env.VERSION, 'beta') || contains(env.VERSION, 'rc') }}
+      # perform production release of artifacts
+      - name: Artifacts
+        run: |
+          echo "pushing release artifacts"
+
```
Comment (hdonnay) [inline .github/workflows/release.yml:60]: This doesn't do anything

**Review by hdonnay — CHANGES_REQUESTED** [review]

File: .chglog/config.yml (hunk 1 above)
Comment (ldelossa) [reply .chglog/config.yml:5]: definitely.

File: .chglog/config.yml (hunk 2 above)
Comment (ldelossa) [reply .chglog/config.yml:18]: You can still link to a github issue in the commit. We are techically supported to link GH issues to jira.

File: .github/workflows/prepare-release.yml (hunk 4 above)
Comment (ldelossa) [reply .github/workflows/prepare-release.yml:5]: yes its wrong.

File: .github/workflows/release.yml (hunk 5 above)
Comment (ldelossa) [reply .github/workflows/release.yml:45]: I can remove it, was there for debuging

**Review by hdonnay — APPROVED** [review]
sgtm

**Pull request merged**